feat: add reference documentation for grafana and prometheus (#57)
charmed-hpc/docs: reference/monitoring/prometheus.md

(reference-monitoring-prometheus)=

> # Prometheus metrics and alerts

Reviewer: Given how similar the sections are, it might make readability easier if this were a table with columns: Charm, upstream link, UI query. And then also have a section with the additional Slurmctld information.

This is an overview of all the charms used in Charmed HPC that provide [Prometheus](https://prometheus.io)
monitoring metrics and alerts.

All metrics and alerts can be viewed from Prometheus or from the [Grafana](https://grafana.com) web interface.
See {ref}`howto-manage-integrate-with-cos` for more information.

## Slurmctld

The `slurmctld` charm provides:

- Resource usage per partition, account or user.
- Metrics about Jobs statuses.
- RPC metrics for `slurmctld`.
- Prometheus Slurm Exporter statistics.

See the `prometheus-slurm-exporter` [upstream documentation](https://github.com/rivosinc/prometheus-slurm-exporter)
for more information on provided metrics.

> You can check its metrics through the Prometheus or Grafana UI using the following query:

Reviewer: ```suggestion
You can search for `slurmctld` metrics through the Prometheus or Grafana UI using the following query:
```
